### PR TITLE
update any cached registries

### PR DESCRIPTION
**Pull request by IanButterworth**
Fixes https://github.com/julia-actions/cache/issues/110

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -132,6 +132,22 @@ runs:
...
 
+    # issue https://github.com/julia-actions/cache/issues/110
+    # Pkg may not run `Registry.update()` if a manifest exists, which may exist because of a
+    # `Pkg.dev` call or because one is added to the repo. So be safe and update cached registries here.
+    # Older (~v1.0) versions of julia that don't have `Pkg.Registry.update()` seem to always update registries in
+    # Pkg operations. So this is only necessary for newer julia versions.
+    - name: Update any cached registries
+      if: ${{ inputs.cache-registries == 'true' }}
+      run: |
+        if [ -d "${{ steps.paths.outputs.depot }}/registries" ] && [ "$(ls -A ${{ steps.paths.outputs.depot }}/registries)" ]; then
```
Comment (omus) [inline action.yml:143]:
```suggestion
        if [ -d "${{ steps.paths.outputs.depot }}/registries" ] && [ -n "$(ls -A ${{ steps.paths.outputs.depot }}/registries)" ]; then
```

**Review by omus — APPROVED** [review]
In an ideal world we'd avoid performing the registry update on up to date registries. We can always add such an optimization after the fact.

**Comment by IanButterworth** [comment]
(I'd merge but approval is required here)

**Review by DilumAluthge — APPROVED** [review]

**Comment by IanButterworth** [comment]
Thanks!

**Pull request merged**